syntax = "proto3";
package key;





/*
UpdateCollectorKeyByName请求
*/
message UpdateCollectorKeyByNameRequest {
message Parent {
    /*
    唯一名称
    */
    string name = 1 ;
}
        
message TagDefine {
    /*
    名字
    */
    string name = 1 ;
    /*
    默认值
    */
    string default = 2 ;
    /*
    是否只读
    */
    bool readOnly = 3 ;
}
        
message ParamDefine {
    /*
    名字
    */
    string name = 1 ;
    /*
    值类型
    */
    string valueType = 2 ;
    /*
    默认值
    */
    string default = 3 ;
    /*
    是否只读
    */
    bool readOnly = 4 ;
}
        
    
    /*
    父Key
    */
    Parent parent = 1 ;
    /*
    唯一名称
    */
    string name = 2 ;
    /*
    key值
    */
    string key = 3 ;
    /*
    key名
    */
    string keyName = 4 ;
    /*
    标签
    */
    repeated string labels = 5 ;
    /*
    类型
    */
    string metricType = 6 ;
    /*
    数据类型
    */
    string dataType = 7 ;
    /*
    所属Agent
    */
    string agentType = 8 ;
    /*
    维度定义
    */
    repeated TagDefine tagDefine = 9 ;
    /*
    参数定义
    */
    repeated ParamDefine paramDefine = 10 ;
    /*
    帮助
    */
    string help = 11 ;
    /*
    单位
    */
    string unit = 12 ;
}

/*
UpdateCollectorKeyByName返回
*/
message UpdateCollectorKeyByNameResponse {
    /*
    指标id
    */
    string keyId = 1 ;
}

/*
UpdateCollectorKeyByNameApi返回
*/
message UpdateCollectorKeyByNameResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    key.UpdateCollectorKeyByNameResponse data = 4 ;
}

